src/device/cuda_intrinsics/synchronization.jl

```diff
@@ -12,6 +13,54 @@ threads in the block.
 """
 @inline sync_threads() = ccall("llvm.nvvm.barrier0", llvmcall, Cvoid, ())
 
+"""
+    sync_threads_count(predicate::Int32)
+
+Identical to `__syncthreads()` with the additional feature that it evaluates predicate
+for all threads of the block and returns the number of threads for which `predicate` evaluates
+to non-zero.
+
+    sync_threads_count(predicate::Bool)
+
+Identical to `__syncthreads()` with the additional feature that it evaluates predicate
+for all threads of the block and returns the number of threads for which `predicate` evaluates
+to `true`.
+"""
+@inline sync_threads_count(predicate::Int32) = ccall("llvm.nvvm.barrier0.popc", llvmcall, Int32, (Int32,), predicate)
+@inline sync_threads_count(predicate::Bool) = sync_threads_count(predicate ? Int32(1) : Int32(0))
```

Comment:
re. friendlier output types, maybe also convert to `Int` here.

Reply:
Isn't `Int` host dependent? I would keep these as `Int32`

Reply:
It is, but we've been doing that for `threadIdx` in the assumption that we'd be able to switch that to a GPU-native type at some point. It also makes it easier for LLVM to reason about integer arithmetic, since most values (literals, indexing, etc) is dealing with CPU-native Int already.